ruby/vampires.rb

```diff
@@ -11,49 +11,64 @@ def detection
 	
 	puts "How old are you? What year were you born? "
 	 age = gets.chomp
-   if age < "100 "
+   if age.to_i < 100
      age_check = true 
    else 
      age_check = false
    end  
 
-	puts "Our company cafeteria serves garlic bread.
-	       Should we order some for you? y/n"
+	puts "Our company cafeteria serves garlic bread.Should we order some for you? yes/no"
 	 garlic = gets.chomp
-   if garlic == "y" 
+   if garlic == "yes" 
      garlic_check = true
-    else 
+   else
      garlic_check = false
-    end
+   end
 
-	puts "Would you like to enroll in the company’s health insurance?y/n"
+	puts "Would you like to enroll in the company’s health insurance? yes/no/waive"
 	 health_insurance = gets.chomp
-	  if health_insurance == "y"
+	  if health_insurance == "yes"
 	     health_insurance_check = true
+	  elsif health_insurance == "waive"
+       health_insurance_check = true
 	  else 
-	   health_insurance_check =false
+	   health_insurance_check = false
 	  end
-	
-	 age_check && garlic_check 
-    puts"Probably not a vampire."
-
-   age_check && garlic_check || health_insurance_check 
-	  puts "Probably a vampire."
-
-   age_check && garlic_check && health_insurance_check 
-    puts "Almost certainly a vampire."
-
-   name_check = false  
- 	  puts "Definitely a vampire"
- 	  
-   print "Results inconclusive."
-
-   puts "Result"
-    name
-    age
-    garlic
-    health_insurance
-      
+	  
+	 puts "What is you allergies? Type done when finished."
+	   allergies = gets.chomp
+	   
+	 until allergies == "done"
+	  if allergies == "sunshine"
+	    puts "Probably a vampire."
```

Comment:
This looks good but I would organize your allergy checking to happen before your vampire results are decided.

An important way to consider designing a workflow is to consider which steps should be taken in which order.

In this case I think a good approach would be to gather all of the user information first and then do you vampire evaluations.